core/src/Cuda/Kokkos_Cuda_Error.hpp

```diff
@@ -27,13 +27,7 @@
 namespace Kokkos {
 namespace Impl {
 
-void cuda_stream_synchronize(
-    const cudaStream_t stream,
-    Kokkos::Tools::Experimental::SpecialSynchronizationCases reason,
-    const std::string& name);
 void cuda_device_synchronize(const std::string& name);
-void cuda_stream_synchronize(const cudaStream_t stream,
-                             const std::string& name);
```

Comment:
Ultimately, these functions don't need to be exposed since they are only used internally. Also note, that the last overload didn't have an implementation.